enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -370,6 +226,30 @@ hardware as the first step toward provisioning that hardware.
 
 [Cluster-API Provider Bare Metal](https://github.com/openshift/cluster-api-provider-baremetal/) will need to gain the ability to interact with the new assisted installer APIs. Specifically it will need to match a Machine with an available assisted Agent that is ready to be provisioned. This will be in addition to matching a Machine with a BareMetalHost, which it already does today.
 
+Other platforms may benefit from similar capability, such as on-premise virtualization
+platforms. Ideally the ability to match a Machine with an Agent will be delivered
+so that it can be integrated into multiple machine-api providers. Each platform would
+perform this workflow:
+
+1. A Machine gets created, usually as a result of a MachineSet scaling up.
+1. The platform does whatever is necessary to boot the discovery live ISO on a host.
+1. The platform waits for a corresponding Agent resource to appear.
+1. The Agent gets associated with the Machine.
+1. The agent-based provisioning workflow is initiated by the machine-api provider.
+
+#### Host Approval
+
+It is important that when an agent checks in, it not be allowed to join a
+cluster until an actor has approved it. From a security standpoint, it is not
+ok for anyone who can access a copy of a discovery ISO and/or access the API
+where agents report themselves to implicitly have the capability to join a
+cluster as a Node.
+
+The host CRD has a field called "Role" that defaults to unset, and it must be
+set prior to provisioning to one of "master", "worker", or "auto-assign". The
+act of assigning a role and a cluster to a host will implicitly approve it to
+join that cluster with the specified role.
```

Comment:
This still allows any arbitrary host to contact the service in the cluster that manages the agents, right? Are there any security implications from that? Can a user of that API learn anything about the cluster that we wouldn't want an unknown actor to know?

Reply:
The API the agent reports to will require authentication, and that authentication will be baked into the live ISO. I don't think the agent can learn any information about anything else through that API today: @avishayt is that right?

Reply:
Did we find the answer to the security question here?

Reply:
@avishayt @filanov WDYT?